tools/update_unicode_iso8859_tables.dart

```diff
@@ -0,0 +1,300 @@
+//
+// Updates the definitions in `lib/src/codepages/unicode_iso8859.g.dart`,
+// between the `// -- BEGIN GENERATED CONTENT --` and
+// `// -- END GENERATED CONTENT --` markers, or inserts such a block at the
+// end of the file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+
+/// The directory of this script. Used as base for other relative paths.
+final String scriptDir = p.dirname(Platform.script.toFilePath());
+
+/// The file containing the generated constants.
+final String tableFilePath = p.join(
+    scriptDir, '..', 'lib', 'src', 'codepages', 'unicode_iso8859.g.dart');
+
+/// The base URI for the Unicode Consortium table files.
+final baseUri = Uri.parse('https://unicode.org/Public/MAPPINGS/ISO8859/');
+
+/// Path to a cache directory used to cache downloaded table files.
+final cachePath = p.join(
+    scriptDir, '..', '.dart_tool', 'iso8859_tool_http_cache', 'unicode.org');
+
+/// Start-marker for generated content in the [tableFilePath] file.
+///
+/// Generation only changes content from this marker to the
+/// [generatedContentTail] end marker.
+const generatedContentLead = '// -- BEGIN GENERATED CONTENT --';
+
+/// End-marker for generated content in the [tableFilePath] file.
+///
+/// Ends region started by [generatedContentLead].
+const generatedContentTail = '// -- END GENERATED CONTENT --';
+
+// Rebuild the tables used for ISO-8859 code pages.
+void main(List<String> args) async {
+  var tableFile = File(tableFilePath);
+  if (!tableFile.existsSync()) {
+    stderr.writeln('Cannot find file to generate into: $tableFile');
+    stderr.writeln('Create file with proper copyright header and run again.');
+  }
+  var tableFileContent = tableFile.readAsStringSync();
+
+  // Generate new declarations.
+  var declarations = await generateTables();
+
+  // Find generated content markers in existing file.
+  var generatedContentMatch = RegExp('$generatedContentLead'
+          r'.*\r?\n(?:// Generated.*\r?\n)?([^]*?\r?\n)'
+          '$generatedContentTail'
+          r'(.*\r?\n)?'
+          r'|$')
+      .firstMatch(tableFileContent)!;
+
+  var existingDeclarations = generatedContentMatch[1] ?? '';
+  if (declarations == existingDeclarations) {
+    // If no change, no write.
+    stderr.writeln('No changes');
+    return;
+  }
+  var replacement = replacementString(declarations);
+  // Guess file's newline, change to '\r\n' if necessary.
+  if (Platform.isWindows && tableFileContent.contains('\r\n')) {
+    replacement = replacement.replaceAll('\n', '\r\n');
+  }
+  tableFileContent = tableFileContent.replaceRange(
+      generatedContentMatch.start, generatedContentMatch.end, replacement);
+  tableFile.writeAsStringSync(tableFileContent);
+  print('Written: $tableFile');
+}
+
+/// Shared HTTP client used by [fetchTable], initialized if needed.
+///
+/// Closed by [generateTables] when done fetching.
+HttpClient? httpClient;
```

Comment:
Could you use `package:http` for this?

Reply:
I could, but adding more dependencies isn't something I do lightly.

In this case, it's a dev-dependency, so it only affects ourselves.
(Also, I don't actually have experience with `package:http`, so I just used what I know.)

Reply:
I don't have strong feelings about this but we are probably going to ask people to not use `HttpClient` directly in the next couple of months so, if we can avoid using it ourselves, that we could ideal.

Reply:
That is a very good reason. And again, it's a dev-dependency, so nobody else should care.

Reply:
What is our reasoning for avoiding `HttpClient` when the code is know to only be used on platforms that  support it?

Reply:
> What is our reasoning for avoiding `HttpClient` when the code is know to only be used on platforms that support it?

Developers will assume that our code reflects best practices so, if we use `HttpClient`, then other people will assume that they should too. In my mind, there are very few reasons to use `HttpClient` directly.

Reply:
IMO keeping core package dependencies as flat as possible should weigh higher than making our code look like recommended app code. I do think that having more sample code will make it less likely authors will need to go looking in packages like this to find a pattern to copy/paste.

Reply:
> IMO keeping core package dependencies as flat as possible should weigh higher than making our code look like recommended app code. I do think that having more sample code will make it less likely authors will need to go looking in packages like this to find a pattern to copy/paste.

I don't feel very strongly about this. The dependency would only be for dev though so it wouldn't impact users.

Reply:
Another possibility is that we remove the download code, and just require the user to have pre-downloaded the files to *somewhere*, so they can point to that directory.

This was just me making my own life easier, when I had to do the same thing multiple times, but it's unnecessary complexity when we can just ask the user to use `curl` first. Like:
```bash
curl -OOOOOOOOOOOOOOO https://unicode.org/Public/MAPPINGS/ISO8859/8859-{1,2,3,4,5,6,7,8,9,10,11,13,14,15,16}.TXT
```

I think I'll just remove the download code. It's safer.